Code/Framework/Tests/InstanceDataHierarchy.cpp

```diff
@@ -1314,4 +1410,243 @@ namespace UnitTest
         run();
     }
 
+    TEST_F(InstanceDataHierarchyGroupTestFixture, TestNormalGroups)
+    {
+        using namespace AzToolsFramework;
+
+        // Setting up the data node hierarchy
+        AZ::SerializeContext serializeContext;
+        serializeContext.CreateEditContext();
+        Entity::Reflect(&serializeContext);
+        GroupTestComponent::Reflect(&serializeContext);
+
+        AZStd::unique_ptr<AZ::Entity> testEntity1(new AZ::Entity());
+        testEntity1->CreateComponent<GroupTestComponent>();
+
+        InstanceDataHierarchy instanceDataHierarchy;
+        instanceDataHierarchy.AddRootInstance(testEntity1.get());
+        instanceDataHierarchy.Build(&serializeContext, 0);
+
+        // Adding the nodes to a node stack
+        auto rootNode = instanceDataHierarchy.GetRootNode();
+        AZStd::stack<InstanceDataNode*> nodeStack;
+        nodeStack.push(rootNode);
+        InstanceDataNode* componentNode1 = nullptr;
+        while (!nodeStack.empty())
+        {
+            InstanceDataNode* node = nodeStack.top();
+            nodeStack.pop();
+            if (node->GetClassMetadata()->m_typeId == AZ::AzTypeInfo<GroupTestComponent>::Uuid())
+            {
+                componentNode1 = node;
+                break;
+            }
+            for (InstanceDataNode& child : node->GetChildren())
+            {
+                nodeStack.push(&child);
+            }
+        }
+        // Iterating through the children in the instance data hierarchy to verify their properties
+        ASSERT_TRUE(componentNode1 != nullptr);
+        for (auto child : componentNode1->GetChildren())
+        {
+            AZStd::string childName(child.GetElementMetadata()->m_name);
+            if (childName.compare("GroupFloat") == 0)
+            {
+                // False for any child node with serializable data
+                ASSERT_FALSE(child.GetElementEditMetadata()->IsClassElement());
+                // Child node should never be a ClassElement::Group, unless it is the root node of a ToggleGroup 
+                ASSERT_NE(child.GetElementEditMetadata()->m_elementId, AZ::Edit::ClassElements::Group);
+                // Ensuring that this node was assigned to the appropriate group
+                ASSERT_EQ(child.GetGroupElementMetadata()->m_description, "Normal Group");
+                // Ensuring that this node has the correct parent
+                ASSERT_EQ(child.GetParent()->GetClassMetadata()->m_name, "GroupTestComponent"); 
+            }
+        }
+    }
+
+    TEST_F(InstanceDataHierarchyGroupTestFixture, TestToggleGroups)
```

Comment:
suggestion: With the naming again, it might be good to say have a test that is called `GroupToggleCanBeExpanded`, and `GroupToggleCanBeCollapsed`, `GroupToggleHidesProperties` etc... When one fails the output is so much more useful. It's hard to do well for sure but can make a massive difference - maybe have a think about how we could break these down a little more